### PR TITLE
Crash on `clear` when looking into scrollback buffer

### PR DESCRIPTION
**Pull request by kchibisov**
Fixes: #3112

cc @nixpulvis @sQVe

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Performance bottleneck when clearing colored rows
 - Vague startup crash messages on Windows with WinPTY backend
 - Deadlock on Windows when closing Alacritty using the title bar "X" button (ConPTY backend)
+- Crash on `clear` when looking into scrollback buffer
```
Comment (chrisduerr) [inline CHANGELOG.md]: `looking into scrollback buffer` might be a bit confusing, maybe the following is a bit better?

```suggestion
- Crash on `clear` when scrolled up in history
```

File: alacritty_terminal/src/grid/mod.rs [hunk 2]
```diff
@@ -544,6 +544,8 @@ impl<T: GridCell + PartialEq + Copy> Grid<T> {
         let positions = self.lines - iter.cur.line;
         let region = Line(0)..self.num_lines();
 
+        // Reset display offset
+        self.display_offset = 0;
         // Clear the viewport
```
Comment (chrisduerr) [inline alacritty_terminal/src/grid/mod.rs]:
```suggestion

         // Clear the viewport
```

File: alacritty_terminal/src/grid/mod.rs [hunk 3]
```diff
@@ -544,7 +544,10 @@ impl<T: GridCell + PartialEq + Copy> Grid<T> {
         let positions = self.lines - iter.cur.line;
         let region = Line(0)..self.num_lines();
 
-        // Clear the viewport
+        // Reset display offset
+        self.display_offset = 0;
+
+         // Clear the viewport
```
Comment (chrisduerr) [inline alacritty_terminal/src/grid/mod.rs]:
```suggestion
        // Clear the viewport
```

Comment (kchibisov) [reply alacritty_terminal/src/grid/mod.rs]: Nah, I was faster

**Pull request merged**